pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -369,18 +475,25 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 
 // Create the CLusterLoadAssignment. At this moment the options must have been applied to the locality lb endpoints.
 func (b *EndpointBuilder) createClusterLoadAssignment(llbOpts []*LocalityEndpoints) *endpoint.ClusterLoadAssignment {
-	llbEndpoints := make([]*endpoint.LocalityLbEndpoints, 0, len(llbOpts))
-	for _, l := range llbOpts {
-		llbEndpoints = append(llbEndpoints, &l.llbEndpoints)
-	}
 	return &endpoint.ClusterLoadAssignment{
 		ClusterName: b.clusterName,
-		Endpoints:   llbEndpoints,
+		Endpoints:   ExtractEnvoyEndpoints(llbOpts),
 	}
 }
 
-func (b *EndpointBuilder) IsDNSCluster() bool {
-	return b.service != nil && (b.service.Resolution == model.DNSLB || b.service.Resolution == model.DNSRoundRobinLB)
+func ExtractEnvoyEndpoints(locEps []*LocalityEndpoints) []*endpoint.LocalityLbEndpoints {
+	llbEndpoints := make([]*endpoint.LocalityLbEndpoints, 0, len(locEps))
+	for _, eps := range locEps {
+		llbEndpoints = append(llbEndpoints, &eps.llbEndpoints)
```

Comment:
do we need `eps := eps` here? I think not since its a pointer to the object inside `eps`, not a pointer to eps...